GAE/src/org/waterforpeople/mapping/app/web/CurrentUserServlet.java

```diff
@@ -83,8 +83,12 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
 
         final VelocityContext context = new VelocityContext();
         final UserDao uDao = new UserDao();
-        final String currentUserEmail = UserServiceFactory.getUserService()
-                .getCurrentUser().getEmail().toLowerCase();
+        final com.google.appengine.api.users.User currentGoogleUser = UserServiceFactory
+                .getUserService().getCurrentUser();
+        if (currentGoogleUser == null) {
```

Comment:
If we're going to `return` there is no point on preparing a template that we're not going to use. I propose to move this check as the first thing to do in the method